src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotAddEmailsToSamGroupStep.java

```diff
@@ -0,0 +1,44 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.service.auth.iam.IamRole;
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.service.job.DefaultUndoStep;
+import bio.terra.service.snapshot.flight.SnapshotWorkingMapKeys;
+import bio.terra.service.snapshotbuilder.SnapshotRequestDao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.List;
+import java.util.UUID;
+
+public class CreateSnapshotAddEmailsToSamGroupStep extends DefaultUndoStep {
+  private final AuthenticatedUserRequest userRequest;
+  private final IamService iamService;
+  private final SnapshotRequestDao snapshotRequestDao;
+  private final UUID snapshotRequestId;
+
+  public CreateSnapshotAddEmailsToSamGroupStep(
+      AuthenticatedUserRequest userRequest,
+      IamService iamService,
+      SnapshotRequestDao snapshotRequestDao,
+      UUID snapshotRequestId) {
+    this.userRequest = userRequest;
+    this.iamService = iamService;
+    this.snapshotRequestDao = snapshotRequestDao;
+    this.snapshotRequestId = snapshotRequestId;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    FlightMap workingMap = context.getWorkingMap();
+    String groupName =
+        workingMap.get(SnapshotWorkingMapKeys.SNAPSHOT_FIRECLOUD_GROUP_NAME, String.class);
+    List<String> emailsToAddToGroup =
+        List.of(snapshotRequestDao.getById(snapshotRequestId).getCreatedBy());
+    iamService.overwriteGroupPolicyEmailsIncludeRequestingUser(
```

Comment:
I think it would be clearer if both emails were retrieved here (so moving the call to getUserInfoAndVerify here) and then added using the existing overwriteGroupPolicyEmails method? What is the motivation to creating and using this new method, overwriteGroupPolicyEmailsIncludeRequestingUser?

Reply:
Yeah, I agree it is not super clear. The main motivation for the new methods is to be able to call getUserInfoAndVerify() from within SamIam. This follows a pattern we use in other resource creation, like`createDatasetResourceRequest` - We pass in the special policies and then grab the steward info and admin info in SamIam. Maybe we can talk about this more in parking lot tomorrow.

Reply:
Okay, maybe we could just add a comment to the step clarifying what it does?

Reply:
(Emerging from lurkdom) if you end up adding a comment to the step, I'd like to suggest adding it as a docstring to the step's constructor.  That way when someone is looking at the flight constructor, when they click on the step in their IDE they'll see its docstring :)